### PR TITLE
{data}[foss/2023a] GDAL v3.7.1

### PR DESCRIPTION
**Pull request by stefan-wolfsheimer**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/g/GDAL/GDAL-3.7.1-foss-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    ('SWIG', '4.1.1'),
+]
+
+# iterative build for both static and shared libraries
+configopts = [
+    "-DGDAL_USE_INTERNAL_LIBS=OFF -DArrow_DIR=$EBROOTARROW",
+    "-DGEOTIFF_INCLUDE_DIR=$EBROOTLIBGEOTIFF/include -DPython_ROOT=$EBROOTPYTHON -DBUILD_SHARED_LIBS=OFF",
+    "-DGDAL_USE_INTERNAL_LIBS=OFF -DArrow_DIR=$EBROOTARROW",
+    "-DGEOTIFF_INCLUDE_DIR=$EBROOTLIBGEOTIFF/include -DPython_ROOT=$EBROOTPYTHON ",
+]
```
Comment (branfosj) [inline easybuild/easyconfigs/g/GDAL/GDAL-3.7.1-foss-2023a.eb:69]: Similar to #18728, this is changing the `configopts` compared to the recent easyconfigs.

**Review by branfosj — CHANGES_REQUESTED** [review]

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bwd-rockylinux-92 - Linux Rocky Linux 9.2 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), Python 3.9.16
See https://gist.github.com/SebastianAchilles/befc1e4d3ad5c7323914e5a2775d5365 for a full test report.

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen2

File: easybuild/easyconfigs/g/GDAL/GDAL-3.7.1-foss-2023a.eb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    ('pkgconf', '1.9.5'),
+    ('Bison', '3.8.2'),
+]
+dependencies = [
+    ('Python', '3.11.3'),
+    ('netCDF', '4.9.2'),
+    ('expat', '2.5.0'),
+    ('GEOS', '3.12.0'),
+    ('SQLite', '3.42.0'),
+    ('libxml2', '2.11.4'),
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/g/GDAL/GDAL-3.7.1-foss-2023a.eb:35]: I think `libarchive` is required to fix `libgdal.so.33: undefined symbol: archive_read_support_format_rar5`

```suggestion
    ('SQLite', '3.42.0'),
    ('libarchive', '3.6.2'),
    ('libxml2', '2.11.4'),
```

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen2

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/g/GDAL/GDAL-3.7.1-foss-2023a.eb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+local_configopts_common += "-DGEOTIFF_INCLUDE_DIR=$EBROOTLIBGEOTIFF/include -DPython_ROOT=$EBROOTPYTHON "
+
+configopts = [
+    local_configopts_common + "-DBUILD_SHARED_LIBS=OFF",
+    local_configopts_common
+]
+
+
+sanity_check_paths = {
+    'files': ['lib/libgdal.a', 'lib/libgdal.so'],
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/g/GDAL/GDAL-3.7.1-foss-2023a.eb:75]:
```suggestion
    'files': ['lib/libgdal.a', 'lib/libgdal.%s' % SHLIB_EXT],
```

**Review by SebastianAchilles — CHANGES_REQUESTED** [review]

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
zen2-ubuntu-2304 - Linux Ubuntu 23.04 (Lunar Lobster), x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.11.2
See https://gist.github.com/SebastianAchilles/269622ad5c2698bbe443aa9b1a9a9778 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bwd-rockylinux-92 - Linux Rocky Linux 9.2 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), Python 3.9.16
See https://gist.github.com/SebastianAchilles/35e779e1a960dd5f734223e04f5c6375 for a full test report.

**Review by SebastianAchilles — APPROVED** [review]
lgtm

**Comment by SebastianAchilles** [comment]
Going in, thanks @stefan-wolfsheimer!

**Pull request merged**